src/err/mod.rs

```diff
@@ -538,12 +538,25 @@ impl PyErr {
     where
         T: ToPyObject,
     {
-        self.is_instance(py, exc.to_object(py).as_ref(py))
+        self.is_instance_bound(py, exc.to_object(py).bind(py).clone())
     }
 
-    /// Returns true if the current exception is instance of `T`.
+    /// Deprecated form of `PyErr::is_instance_bound`.
+    #[cfg_attr(
+        not(feature = "gil-refs"),
+        deprecated(
+            since = "0.21.0",
+            note = "`PyErr::is_instance` will be replaced by `PyErr::is_instance_bound` in a future PyO3 version"
+        )
+    )]
     #[inline]
     pub fn is_instance(&self, py: Python<'_>, ty: &PyAny) -> bool {
+        self.is_instance_bound(py, ty.into_py(py).into_bound(py))
```

Comment:
With `&Bound` for the argument, this should work:

```suggestion
        self.is_instance_bound(py, &ty.as_borrowed())
```

Potentially are you willing to open a PR to recommend using `as_borrowed()` as the way to go from `&PyAny` to `&Bound` in the migration guide? e.g.

```rust
let gil_ref: &PyAny = ...;
let bound: &Bound<PyAny> = &gil_ref.as_borrowed();
```

(hopefully most user code won't need to care about this because all of it will just get migrated over, but this might still be useful for users to be aware as they rewrite their code...)